File: lab-2.2-ByLiza/C#/NumberString.cs
namespace String  
{
    public class NumbersString : BaseString
    {
        public NumbersString(string input) : base(input) { }  
 
        public override int GetLength()
        {
            return _value.Length;
        }

        public override string Shift()
        {
            if (_value.Length > 1)
            {
                char[] chars = _value.ToCharArray();

                char lastChar = chars[chars.Length - 1];
                for (int i = chars.Length - 1; i > 0; i--)
                {
                    chars[i] = chars[i - 1];  
                }
                chars[0] = lastChar; 

                return new string(chars);
            }
            return _value;
        }
    }
}

File: lab-2.2-ByLiza/C++/NumberString.cpp
#include "NumbersString.h"

namespace String {

    NumbersString::NumbersString(const std::string& input) : BaseString(input) {}

    int NumbersString::GetLength() const {
        return _value.length();
    }

    std::string NumbersString::Shift() const {
        if (_value.length() > 1) {
            std::string chars = _value;
            char lastChar = chars[chars.length() - 1];
            for (size_t i = chars.length() - 1; i > 0; --i) {
                chars[i] = chars[i - 1];
            }
            chars[0] = lastChar;
            return chars;
        }
        return _value;
    }

}